plugins/org.eclipse.n4js/src/org/eclipse/n4js/resource/N4JSCrossReferenceComputer.java

```diff
@@ -58,25 +62,43 @@
 	 * @param acceptor
 	 *            the logic that collects the passed EObject found in a cross reference
 	 */
-	void computeCrossRefs(Resource resource, IAcceptor<EObject> acceptor) {
-		TreeIterator<EObject> allContentsIter = resource.getAllContents();
+	public void computeCrossRefs(Resource resource, IAcceptor<ImmutablePair<EObject, EObject>> acceptor) {
+		TreeIterator<EObject> allContentsIter;
+		if (resource instanceof N4JSResource) {
+			allContentsIter = ((N4JSResource) resource).getAllContentIgnoreCachedElements();
+		} else {
+			allContentsIter = resource.getAllContents();
```

Comment:
Hmm, I think you are right. I don't see any reason for traversing the `TModule` either...